src/Plug.vala

```diff
@@ -378,20 +378,22 @@ namespace Power {
         private static bool lid_detect () {
             var interface_path = File.new_for_path ("/proc/acpi/button/lid/");
 
-            try {
-                var enumerator = interface_path.enumerate_children (
-                GLib.FileAttribute.STANDARD_NAME,
-                FileQueryInfoFlags.NONE);
-                FileInfo lid;
-                if ((lid = enumerator.next_file ()) != null) {
-                    debug ("Detected lid switch");
-                    return true;
-                }
+            if (interface_path.query_exists ()) {
```

Comment:
use upower instead proc file system. /proc is deprecated and shouldn't be used anymore https://upower.freedesktop.org/docs/UPower.html
see properties

Reply:
OK, I'll try this https://upower.freedesktop.org/docs/UPower.html#UPower:LidIsPresent